echopype/convert/api.py

```diff
@@ -0,0 +1,553 @@
+            io.check_file_permissions(fsmap)
+            if isinstance(fs, LocalFileSystem):
+                # Only make directory if local file system
+                # otherwise it will just create the object path
+                fs.mkdir(fsmap.root)
+        except FileNotFoundError:
+            raise ValueError("Specified save_path is not valid.")
+
+    return out_path  # output_path is always a list
+
+
+def to_file(
+    echodata: EchoData,
+    engine,
+    save_path=None,
+    compress=True,
+    overwrite=False,
+    parallel=False,
+    output_storage_options={},
+    **kwargs,
+):
+    """Save content of EchoData to netCDF or zarr.
+
+    Parameters
+    ----------
+    engine : str {'netcdf4', 'zarr'}
+        type of converted file
+    save_path : str
+        path that converted .nc file will be saved
+    compress : bool
+        whether or not to perform compression on data variables
+        Defaults to ``True``
+    overwrite : bool
+        whether or not to overwrite existing files
+        Defaults to ``False``
+    parallel : bool
+        whether or not to use parallel processing. (Not yet implemented)
+    output_storage_options : dict
+        Additional keywords to pass to the filesystem class.
+    """
+    # TODO: revise below since only need 1 output file in use case EchoData.to_zarr()/to_netcdf()
+    if parallel:
+        raise NotImplementedError(
+            "Parallel conversion is not yet implemented."
+        )
+    if engine not in XARRAY_ENGINE_MAP.values():
+        raise ValueError("Unknown type to convert file to!")
+
+    # Assemble output file names and path
+    format_mapping = dict(map(reversed, XARRAY_ENGINE_MAP.items()))
+    output_file = _validate_path(
+        source_file=[echodata.source_file],
+        file_format=format_mapping[engine],
+        save_path=save_path,
+    )
+
+    # Get all existing files
+    exist_list = []
+    fs = fsspec.get_mapper(
+        output_file[0], **output_storage_options
+    ).fs  # get file system
+    for out_f in output_file:
+        if fs.exists(out_f):
+            exist_list.append(out_f)
+
+    # Sequential or parallel conversion
+    for src_f, out_f in zip([echodata.source_file], output_file):
+        if out_f in exist_list and not overwrite:
+            print(
+                f"{dt.now().strftime('%H:%M:%S')}  {src_f} has already been converted to {engine}. "
+                f"File saving not executed."
+            )
+            continue
+        else:
+            if out_f in exist_list:
+                print(f"{dt.now().strftime('%H:%M:%S')}  overwriting {out_f}")
+            else:
+                print(f"{dt.now().strftime('%H:%M:%S')}  saving {out_f}")
+            _save_groups_to_file(
+                echodata,
+                output_path=_normalize_path(
+                    out_f, engine, output_storage_options
+                ),
+                engine=engine,
+                compress=compress,
+            )
```

Comment:
I am thinking we can make `to_file` to only save groups to a single file (i.e. can take out the loops). This is because under our new intended use of EchoData object, regardless of whether it is created from `open_raw` or `open_mfraw`, each attribute will only hold 1 xr.Dataset, and there always will only be 1 output file. Users can still parallelize using `open_raw`.

The `parallel` option though I think should stay (though remains not implemented), because we can parallelize the saving of the attributes/groups to the same output file. This would probably become handy once we add in `open_mfraw` since the data held in memory can be big.